source/extensions/transport_sockets/s2a/grpc_tsi.h

```diff
@@ -10,7 +10,13 @@
 #pragma GCC diagnostic ignored "-Wold-style-cast"
 #endif
 
-
+#include "grpc/grpc_security.h"
```

Comment:
Why are all of these includes needed in this file, when they're not used by this file? Now that we've resolved the BUILD issues, can we remove these?

Reply:
Even though they are not used by this file, other files in the s2a directory depend on these imports transitively. This is how the alts extension handles its imports for its various components, so I wanted to mimic the existing style.

Reply:
Can we change the includes so that each file includes everything that it needs? The way the ALTS extension did this turned out to be quite brittle, as we noticed while debugging (and we should probably send a PR that changes the ALTS extension).